contributing/COMMITS.md

```diff
@@ -0,0 +1,242 @@
+made. The commits shown below were easily rearranged into the preferred
+sequence and granularity for review using an interactive git rebase.
+This allowed the 'end-to-end first' development approach to be used,
+without compromising the functionality of the release branch post merge.
+
+```
+[CLI] Add e2e test for store verbs
+[CLI] Add store verb commands
+[CLI] Add stores verb app service entry points
+[API] Add `/stores` to swagger spec
+[API] Add `/stores` e2e tests
+[API] Add `/stores` endpoint and handlers
+[API] Add `/stores` domain logic
+[API] Add `/stores` in mem repo
+[API] Add `/stores` sqlite repo
+[CLI] Connect store app service to API
+```
+
+> NOTE: If you do choose to use this approach, it is important to ensure
+> the full test suite passes at each final merge commit (post-squash)
+> after re-ordering.
+
+#### Making titles easier to read during review
+
+It's ok within a Pull Request to use other titles for squash commits to
+make keeping track of development easier. As long as it is still clear
+to the reviewer which commits will be squashed, and which are intended
+to be merged as-is. Use the `squash` prefix without the `!` to make it
+clear these need to be manually managed.
+
+The following example shows a situation where due to the number of
+changes required, the `--autosquash` format was avoided to make it
+easier to understand:
+
+```
+[API] Add `/replica` endpoint to swagger spec
+squash [API] Add tests for `/replica` endpoint
+squash [API] Add endpoint and handlers
+squash [API] Refactor replica repository interface
+squash [API] Add domain logic
+squash [API] Update in-mem repository implementation
+squash [API] Add sqlite repository support
+```
+
+### Larger projects
+
+On some occasions, such as for much larger pieces of work, we may adopt
+one or more techniques to make the integration of complex work into
+`main` more manageable.
+
+Most strategies are a trade-off. Whichever approach is taken, the
+proposed work process should be agreed by the team before work starts.
+This ensures that system boundaries remain intact whilst work is under
+way, and the division of work and general approach makes sense.
+
+
+#### Merging inside-out
+
+The inside-out approach can allow more frequent integration of work,
+where feature toggles are not available, or add unnecessary complexity.
+Rather than starting by merging end-to-end tests that require the entire
+system to function - forcing the entirety of the work to be in that one
+Merge Request - we break it down into discreet pieces of work, that
+leave public boundaries intact at each stage.
+
+End-to-end tests and other outer-layer acceptance tests can still be
+written ahead of time, they are just kept in a separate development
+branch until, they are ready to be integrated.
+
+This might mean that a specific Pull Request is only covered by other
+kinds of tests, but this is OK if it has been properly coordinated.
```

Comment:
[consider]  If I interpreted this right:
```suggestion
This might mean that a specific Pull Request in isolation is missing
end-to-end tests, but this is OK if it has been properly coordinated.
```